File: app/src/main/java/codes/wokstym/cookingrecipes/models/Unit.java
package codes.wokstym.cookingrecipes.models;

import com.fasterxml.jackson.annotation.JsonValue;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
@AllArgsConstructor
public enum Unit {
    GRAM("gram"),
    SPOON("łyżek"),
    PIECE("sztuk"),
    ML("ml"),
    HANDFUL("garść")
    ;

    @JsonValue
    private final String name;
}

File: app/src/main/java/codes/wokstym/cookingrecipes/models/Unit.kt
package codes.wokstym.cookingrecipes.models

import androidx.annotation.StringRes
import codes.wokstym.cookingrecipes.R

enum class Unit(@StringRes val nameResourceId: Int) {
    GRAM(R.string.gram),
    SPOON(R.string.spoon),
    PIECE(R.string.piece),
    ML(R.string.ml),
    HANDFUL(R.string.handful)
}